playbooks/roles/consul/tasks/acl.yml

```diff
@@ -0,0 +1,28 @@
+---
+
+- name: Install Consul ACL apt requirements
+  apt:
+    name:
+      - python3-pip
+
+- name: Install Consul ACL python requirements
+  pip:
+    executable: pip3
+    name:
+      - python-consul==1.1.0
+      - pyhcl==0.3.10
+
+- name: Configure Consul ACL anonymous token
+  consul_acl:
+    token: anonymous
+    mgmt_token: "{{ consul_acl_master_token }}"
+    rules: "{{ consul_acl_anonymous_policy }}"
+  when: consul_server and consul_acl_anonymous_policy is not none
+
+- name: Create Consul ACL token
+  consul_acl:
+    name: "{{ consul_nodename }}"
+    token: "{{ consul_acl_token }}"
+    mgmt_token: "{{ consul_acl_master_token }}"
+    rules: "{{ consul_acl_policy }}"
+  when: consul_acl_token != "" and consul_acl_policy is not none
```

Comment:
The management token is technically available inside of the Consul server configuration already, but yes I think because `python-consul` runs inside of the target host, it needs the management token for the time it needs to run this task. It will use it to create a new token with the ID specified by `consul_acl_token`, attach a policy to it specified by `rules`, and return. If the token already exists with the same policy, the task does nothing.

Reply:
Since we are currently only sending the management token to the load balancers and the cert manager, I think this is fine for now – these hosts have all the permissions anyway.  If we want to implement further restrictions for the Open edX instances, we should probably let Ocim do this directly, instead of sending a management token to every instance during deployment.  So all good for now, but we should be careful in the future.

Reply:
Yep with the most recent changes though this is a lot more secure - we will only send management tokens to the HAProxy servers when running the new `consul-acl` playbook.

For Ocim, we'll have to give it a separate management token that it just uses on the Ocim VM itself to generate tokens for each Open edX VM. It will then forward that token to the Open edX VM.